Index: src/leetcode/problems/array/medium/RevealCardIncreasingOrder.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/leetcode/problems/array/medium/RevealCardIncreasingOrder.kt	(date 1594534144709)
+++ src/leetcode/problems/array/medium/RevealCardIncreasingOrder.kt	(date 1594534144709)
@@ -0,0 +1,19 @@
+package leetcode.problems.array.medium
+
+import java.util.*
+
+// https://leetcode.com/problems/reveal-cards-in-increasing-order/
+class RevealCardIncreasingOrder {
+
+    fun deckRevealedIncreasing(deck: IntArray): IntArray {
+        val result = IntArray(deck.size)
+        deck.sort()
+        val queue = LinkedList<Int>()
+        (deck.indices).forEach { queue.add(it) }
+        deck.forEachIndexed { index, value ->
+            result[queue.poll()] = value
+            queue.add(queue.poll())
+        }
+        return result
+    }
+}
\ No newline at end of file
